api-reference/_hidden/GridBaseColumn/filterOperations.md

```diff
@@ -6,10 +6,10 @@ default: undefined
 ---
 ---
 ##### shortDescription
-Specifies a set of available filter operations. Applies only if [filterRow](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/filterRow '{basewidgetpath}/Configuration/filterRow/').**visible** and [allowFiltering](/api-reference/_hidden/GridBaseColumn/allowFiltering.md '{basewidgetpath}/Configuration/columns/#allowFiltering') are **true**.
+Applies only if [allowFiltering](/api-reference/_hidden/GridBaseColumn/allowFiltering.md '{basewidgetpath}/Configuration/columns/#allowFiltering') and at least one of the [filterRow](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/filterRow '{basewidgetpath}/Configuration/filterRow/').**visible** and [filterPanel]({basewidgetpath}/Configuration/filterPanel/).**visible** options are **true**.
 
 ---
-The following table lists available filters by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType'). The same filters are assigned to columns of a specific data type by default.
+The following table lists available filters by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType') that can be used both in the filter row and nested [filterBuilder](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#filterBuilder) component. The same filters are assigned to columns of a specific data type by default.
```

Comment:
> that can be used both in the filter row and nested [filterBuilder](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#filterBuilder) component.

I didn't understand this part. Are data types can be used or filters? What do mean by "used"? Can we be more specific?

> The same filters are assigned to columns of a specific data type by default.

"By default" is an outlaw, we better rewrite this sentence while we're at it.

Reply:
I suggest the followng:

1. The following table lists filter operations by [data types](). Users can apply these filters both in the filter row and nested [filterBuilder]() component. 

I would say that use of "by default" was justified in the next sentence, because this phrase is much more concise than the description I can make up:

2. **DataGrid** automatically assigns these filters operations to columns of a specific data type.
2.1 My second thought was the following: 'When **DataGrid** is initialized, it automatically assigns...'

But both these sentences seem a bit clumsy to me. Could you please share your thoughts?

Reply:
```suggestion
The following table lists available filter operations by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType'). Users can apply these filter operations both in the filter row and nested [filterBuilder]({basewidgetpath}/Configuration/#filterBuilder) component. The same filters are assigned to columns of a specific data type by default.
```